sdk/monitor/Azure.Monitor.Query/TroubleshootingGuide.md

```diff
@@ -0,0 +1,184 @@
+
+## Troubleshooting logs query
+
+### Troubleshooting insufficient access error for logs query
+
+If you get an HTTP error with status code 403 (Forbidden), it means the provided credentials doesn't have sufficient permissions to query the workspace.
+
+```text
+"{"error":{"message":"The provided credentials have insufficient access to perform the requested operation","code":"InsufficientAccessError","correlationId":""}}"
+```
+
+1. Check that the application or user that is making the request has sufficient permissions:
+    * You can refer to this document to [manage access to workspaces](https://docs.microsoft.com/azure/azure-monitor/logs/manage-access#manage-access-using-workspace-permissions)
+2. If the user or application is granted sufficient privileges to query the workspace, make sure you're authenticating as that user/application. If you're authenticating using the [DefaultAzureCredential](https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/identity/Azure.Identity), check the logs to verify the credential used is the one you expected. To enable logging, see the [Enable client logging](#enable-client-logging) section.
+
+For more help on troubleshooting authentication errors please see the Azure Identity client library [troubleshooting guide](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.Query/TroubleshootingGuide.md).
+
+### Troubleshooting invalid Kusto query
+
+If you get an HTTP error with status code 400 (Bad Request), you may have an error in your Kusto query. You'll see an error message similar to the one below.
+
+```text
+(BadArgumentError) The request had some invalid properties
+Code: BadArgumentError
+Message: The request had some invalid properties
+Inner error: {
+    "code": "SemanticError",
+    "message": "A semantic error occurred.",
+    "innererror": {
+        "code": "SEM0100",
+        "message": "'take' operator: Failed to resolve table or column expression named 'Appquests'"
+    }
+}
+```
+
+The error message in `innererror` may include the where the Kusto query has an error. You may also refer to the [Kusto Query Language](https://docs.microsoft.com/azure/data-explorer/kusto/query) (KQL) reference docs to learn more about querying logs using KQL.
+
+### Troubleshooting empty log query results
+
+If your Kusto query returns empty or has no logs, please validate the following:
+
+- You have the right workspace ID
+- You're setting the correct time interval for the query. Try expanding the time interval for your query to see if that returns any results.
+- If your Kusto query also has a time interval, the query is evaluated for the intersection of the time interval in the query string and the time interval set in the `timespan` param provided the query API. The intersection of these time intervals may not have any logs. To avoid any confusion, it's recommended to remove any time interval in the Kusto query string and use `timespan` explicitly.
```

Comment:
```suggestion
- If your Kusto query also has a time interval, the query is evaluated for the intersection of the time interval in the query string and the time interval set in the `timeRange` parameter provided the query API. The intersection of these time intervals may not have any logs. To avoid any confusion, it's recommended to remove any time interval in the Kusto query string and use `timeRange` explicitly.
```